### PR TITLE
Move FluidDensityAux from Pronghorn to fluid property module.

### LINKED ISSUE #17546: Move FluidDensityAux from Pronghorn to fluid properties
## Reason
Pronghorn has an auxkernel, `FluidDensityAux`, that computes density from a `SinglePhaseFluidProperties` user object as a function of pressure and temperature. I would like similar capabilities for Cardinal, so it makes sense to move this very general capability up into the fluid properties module.

## Design
Move `FluidDensityAux` and its tests from Pronghorn to the fluid properties module. 

## Impact
Potentially name collisions with apps that have their own `FluidDensityAux`, or changes in parameter names depending on how people defined the pressure and temperature coupled variables.

### PR DESCRIPTION
**Pull request by aprilnovak**
Moved `FluidDensityAux` from Pronghorn to fluid property module.

Closes #17546

### CODE REVIEW

**Comment by loganharbour** [comment]
> This will require updates to Pronghorn input files because we used `pressure` and `T_fluid` for the fluid density, rather than the fluid-property convention of `p` and `T`. Do I open a PR in Pronghorn with those changes that also has the MOOSE submodule update at the same time?

Yep. Make sure it works locally. As soon as we merge into next (we typically coordinate over this), you'll add a submodule update to your Pronghorn PR that is the new merge commit into next. Get that in, then we'll go through and invalidate the failure on next.

File: modules/fluid_properties/test/tests/auxkernels/tests [hunk 1]
```diff
@@ -17,4 +17,13 @@
     exodiff = 'stagnation_temperature_aux_out.e'
     threading = '!pthreads'
   [../]
+  [./fluid_density_aux]
+    type = 'Exodiff'
+    input = 'fluid_density_aux.i'
+    exodiff = 'fluid_density_aux_out.e'
+    threading = '!pthreads'
```
Comment (loganharbour) [inline modules/fluid_properties/test/tests/auxkernels/tests:24]:
```suggestion
```

You shouldn't need this now, right?

File: modules/fluid_properties/test/tests/auxkernels/tests [hunk 2]
```diff
@@ -17,4 +17,13 @@
...
+    requirement = "The system shall compute fluid density from pressure and temperature."
```
Comment (loganharbour) [inline modules/fluid_properties/test/tests/auxkernels/tests:25]:
```suggestion
    requirement = "The system shall have the ability to compute the fluid density from pressure and temperature."
```

**Review by loganharbour — CHANGES_REQUESTED** [review]

File: modules/fluid_properties/test/tests/auxkernels/tests (hunk 1 above)
Comment (aprilnovak) [reply modules/fluid_properties/test/tests/auxkernels/tests:24]: I don't actually know why that's there, I just copied the other tests in that file - does the fluid property app only work for certain threading options? But if I can remove it, I'll do that!

**Review by loganharbour — DISMISSED** [review]

File: modules/fluid_properties/test/tests/auxkernels/tests (hunk 1 above)
Comment (aprilnovak) [reply modules/fluid_properties/test/tests/auxkernels/tests]: Whoops, sorry I missed that - had it on my to-do list and it got buried. Should be good now

Comment (loganharbour) [reply modules/fluid_properties/test/tests/auxkernels/tests]: Now that I look at it, all of the similar tests above have threading disabled. I suspect this won't work but we'll see 😆

**Review by loganharbour — APPROVED** [review]

**Pull request merged**